### PR TITLE
feat: Remove `x/wasm` module

### PR DESCRIPTION
**Pull request by zemyblue**
## Description

Detach `x/wasm` module to other repository. [line/wasmd](https://github.com/line/wasmd)

## Motivation and context

We have had a difficult to bump up the upstream cosmwasm/wasmd in lbm-sdk. Because the commit history is difference and the references git is difference, we cannot use merely git merge. So we need to detach `x/wasm` module like `cosmwasm/wasmd`. 

## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/line/lbm-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/line/lbm-sdk/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [x] I have added tests to cover my changes.
- [x] I have updated the documentation accordingly.
- [x] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

**Review by zemyblue — COMMENTED** [review]
I cannot add unittest below two parts to indicate no coverage. Because this is IBC integrate test code and it is tested other integration test. 
* https://github.com/line/lbm-sdk/pull/850/files#annotation_7645931682
* https://github.com/line/lbm-sdk/pull/850/files#annotation_7645931685

File: simapp/app_test.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ import (
 func TestSimAppExportAndBlockedAddrs(t *testing.T) {
 	encCfg := MakeTestEncodingConfig()
 	db := dbm.NewMemDB()
-	app := NewSimApp(log.NewOCLogger(log.NewSyncWriter(os.Stdout)), db, nil, true, map[int64]bool{}, DefaultNodeHome, 0, encCfg, EmptyAppOptions{}, nil)
+	app := NewSimApp(log.NewOCLogger(log.NewSyncWriter(os.Stdout)), db, nil, true, map[int64]bool{}, DefaultNodeHome, 0, encCfg, EmptyAppOptions{} /*, nil*/)
```
Comment (dudong2) [inline simapp/app_test.go:46]:
```suggestion
	app := NewSimApp(log.NewOCLogger(log.NewSyncWriter(os.Stdout)), db, nil, true, map[int64]bool{}, DefaultNodeHome, 0, encCfg, EmptyAppOptions{})
```

File: simapp/app_test.go [hunk 2]
```diff
@@ -64,7 +64,7 @@ func TestSimAppExportAndBlockedAddrs(t *testing.T) {
 	app.Commit()
 
 	// Making a new app object with the db, so that initchain hasn't been called
-	app2 := NewSimApp(log.NewOCLogger(log.NewSyncWriter(os.Stdout)), db, nil, true, map[int64]bool{}, DefaultNodeHome, 0, encCfg, EmptyAppOptions{}, nil)
+	app2 := NewSimApp(log.NewOCLogger(log.NewSyncWriter(os.Stdout)), db, nil, true, map[int64]bool{}, DefaultNodeHome, 0, encCfg, EmptyAppOptions{} /*, nil*/)
```
Comment (dudong2) [inline simapp/app_test.go:67]:
```suggestion
	app2 := NewSimApp(log.NewOCLogger(log.NewSyncWriter(os.Stdout)), db, nil, true, map[int64]bool{}, DefaultNodeHome, 0, encCfg, EmptyAppOptions{})
```

**Review by dudong2 — APPROVED** [review]

**Review by tkxkd0159 — APPROVED** [review]
LGTM

**Pull request merged**